erts/emulator/beam/erl_debugger.c

```diff
@@ -550,36 +717,29 @@ stack_frame_fun_info(Process *c_p, ErtsCodePtr pc, Process *rp, int is_return_ad
 static Eterm
 make_value_or_too_large_tuple(Process *p, Eterm val, Uint max_size) {
     Uint val_size;
-    int tup_arity = 2;
+    const int tup2_sz = 3;
     Eterm result, *hp;
 
-    hp = HAlloc(p, tup_arity + 1);
-    result = make_tuple(hp);
-    *hp++ = make_arityval(tup_arity);
-
+    hp = HAlloc(p, tup2_sz);
     val_size = size_object(val);
     if (val_size <= max_size) {
-        *hp++ = am_value;
-        *hp++ = copy_object(val, p);
+        result = TUPLE2(hp, am_value, copy_object(val, p));
     } else {
-        *hp++ = ERTS_MAKE_AM("too_large");
-        *hp++ = make_small(val_size);
+        result = TUPLE2(hp, ERTS_MAKE_AM("too_large"), make_small(val_size));
     }
 
     return result;
 }
 
 static Eterm
 make_catch_tuple(Process *c_p, ErtsCodePtr catch_addr) {
-    int tup_arity = 2;
+    const int tup2_sz = 3;
     Eterm result, *hp;
 
-    hp = HAlloc(c_p, tup_arity + 1);
-    result = make_tuple(hp);
-    *hp++ = make_arityval(tup_arity);
-
-    *hp++ = ERTS_MAKE_AM("catch");
-    *hp++ = stack_frame_fun_info(c_p, catch_addr, NULL, 0);
+    hp = HAlloc(c_p, tup2_sz);
+    result = TUPLE2(hp,
+                    ERTS_MAKE_AM("catch"),
```

Comment:
Nitpick: `'catch'` is already defined in the standard library, we may as well add it to `atom.names` :-)